### DISCUSSION TITLE
Autocomplete for `tuple[Literal[...]]` in `__getitem__`

### DISCUSSION THREAD

**Discussion by kc0506** [microsoft/pylance-release#6250] [Enhancement]
### Description
When indexing an instance with a `tuple`, provide autocomplete options as if using a normal `Literal` type.

```py
class A:
    def __getitem__(self, key: tuple[Literal["a", "b"], ...]):
        pass
        
A()["a",]  # autocomplete for "a"
```

### Current behavior

- When using `tuple[Literal]` on normal functions, the autocomplete works.
![螢幕擷取畫面 2024-08-09 220519](https://github.com/user-attachments/assets/6204bdce-b99c-4c42-a82f-78f78b4e9c85)

- When used `Literal` alone on `__getitem__`, it also works.
![螢幕擷取畫面 2024-08-09 220618](https://github.com/user-attachments/assets/1601dadd-9256-4d77-81f9-81aa6599bc8f)

- When combining together, it does not work. (But the type checking still works if the type is not correct.)
![螢幕擷取畫面 2024-08-09 220457](https://github.com/user-attachments/assets/99f4341d-bd51-401e-8dd3-6a90874c19db)

### Example use case

It can provide convenience for dataframe-like objects.

```py

class A[T, R]:
    def __init__(self, items: dict[T, R]) -> None:
        self.items = items

    def __getitem__(self, key: tuple[T, ...]) -> tuple[R, ...]:
        return tuple(self.items[k] for k in key)


class B(A[Literal["a", "b"], int]):
    pass


b = B({"a": 1, "b": 2})
b["a", "b"]  # (1, 2)
```

**Comment by bschnurr** [comment]
Thank you for the report. 

I think is is more of an enhancement request. I'll move it to discussions for upvotes.
